lib/fastlane/plugin/wpmreleasetoolkit/actions/common/create_release_backmerge_pull_request_action.rb

```diff
@@ -0,0 +1,189 @@
+                                end
+
+        final_target_branches.map do |target_branch|
+          Fastlane::Helper::GitHelper.checkout_and_pull(source_branch)
+
+          create_backmerge_pr(
+            token: token,
+            repository: repository,
+            title: "Merge #{source_branch} into #{target_branch}",
+            head_branch: source_branch,
+            base_branch: target_branch,
+            labels: labels,
+            milestone: target_milestone&.number,
+            reviewers: reviewers,
+            team_reviewers: team_reviewers
+          )
+        end
+      end
+
+      # Determines the target branches for a release version.
+      #
+      # @param source_release_version [String] the release version to compare against other release branches.
+      # @param default_branch [String] the default branch to use if no target branches are found.
+      # @return [Array<String>] the list of target branches greater than the release version.
+      def self.determine_target_branches(source_release_version:, default_branch:)
+        release_branches = Actions.sh('git', 'branch', '-r', '-l', 'origin/release/*').chomp.split("\n")
+
+        all_release_branches_versions = release_branches
+                                        .map { |branch| branch.match(%r{origin/release/([0-9.]*)})&.captures&.first }
+                                        .compact
+
+        target_branches = all_release_branches_versions.select { |branch| Gem::Version.new(branch) > Gem::Version.new(source_release_version) }
+                                                       .map { |v| "release/#{v}" }
+        target_branches = [default_branch] if target_branches.empty?
+
+        target_branches
+      end
+
+      # Creates a backmerge pull request using the `create_pull_request` Fastlane Action.
+      #
+      # @param token [String] the GitHub token for authentication.
+      # @param repository [String] the repository where the pull request will be created.
+      # @param title [String] the title of the pull request.
+      # @param head_branch [String] the source branch for the pull request.
+      # @param base_branch [String] the target branch for the pull request.
+      # @param labels [Array<String>] the labels to add to the pull request.
+      # @param milestone [String] the milestone to associate with the pull request.
+      # @param reviewers [Array<String>] the individual reviewers for the pull request.
+      # @param team_reviewers [Array<String>] the team reviewers for the pull request.
+      # @return [void]
+      def self.create_backmerge_pr(token:, repository:, title:, head_branch:, base_branch:, labels:, milestone:, reviewers:, team_reviewers:)
+        intermediate_branch = "merge/#{head_branch.gsub('/', '-')}-into-#{base_branch.gsub('/', '-')}"
+        Fastlane::Helper::GitHelper.create_branch(intermediate_branch)
+
+        other_action.push_to_git_remote(tags: false)
```

Comment:
We might want to add a `ConfigItem` of `type: Proc` (hope that's allowed by Fastlane? I don't remember even having seen an action taking a block as parameter so far…) that would be called here if not nil and would allow us to add additional commits to the intermediate branch before finalizing the PR.

The typical use case being for Tumblr, where we'd want to pass a block that restores the version files to their content on the target branch. Though I could imagine some other projects where a similar commit to prevent or anticipate-and-fix conflicts would be useful on other files too.

Reply:
TIL, Fastlane does support Proc!

Everything went fine until I hit this issue and realize having Proc parameters given our `spec_helper.rb` is a bit more complicated: https://github.com/wordpress-mobile/release-toolkit/blob/476f94f8cf02847cab18d88e59270c3b0372999f/spec/spec_helper.rb#L73

The lane it generates in the method above looks like this on runtime, and the Proc completely breaks it 😅:
```ruby
lane :test do
  create_release_backmerge_pull_request(
    {:github_token=>"ghp_fake_token", :repository=>"repo-test/project-test", :source_branch=>"release/30.6", :default_branch=>"main", :intermediate_branch_created_callback=>#<Proc:0x00000001158d5200 <etc...>/create_release_backmerge_pull_request_spec.rb:316>}
  )
end
```

Reply:
Ah yeah I'm not sure there's an easy way to make that Spec Helper work with `Proc` indeed. You might need to write the test lane manually for that one… or alternatively pass a `String` with the `Proc`'s intended source code when calling the stub (so that its `inspect` echoes that source code and that's what gets evaluated by ``Fastlane::Fastfile.new.parse(…)`), instead of passing the actual `Proc` 🤔

Reply:
Implemented here: 6a3204f72fde2c1181bd1a4ee6088a555d678f08